### PR TITLE
fix issue when BCPROP is not set initially

### PR DESCRIPTION
**Pull request by totto82**
In the current code BCPROP is ignored if it is not set in the first time step. This PR make it possible to set BCPROP later in the schedule.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

**Comment by totto82** [comment]
The test failure is expected. The current code does not impose the correct bc for the test case. This fixes a bug, but since there is simple workaround (just adding the BCPROP for the initial step), I dont think it needs to be part of the release. Hopefully it can be merged when the release is done.

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

File: opm/models/discretization/common/tpfalinearizer.hh [hunk 1]
```diff
@@ -469,23 +469,24 @@ private:
...
-                                                     exFluidState.pvtRegionIndex(),
-                                                     bfIndex,
-                                                     bf.area(),
-                                                     bf.integrationPos()[dimWorld - 1],
-                                                     exFluidState};
-                        boundaryInfo_.push_back({myIdx, dir_id, bcdata});
-                    }
+                    const auto& exFluidState = problem_().boundaryFluidState(myIdx, dir_id);
+                    BoundaryConditionData bcdata{type,
+                                                    massrate,
```
Comment (atgeirr) [inline opm/models/discretization/common/tpfalinearizer.hh:483]: Indent.

**Review by atgeirr — CHANGES_REQUESTED** [review]
While I think this is correct, it will likely cause some performance degradation. With this we will always, in every assemble, iterate over all boundary faces even if there are no boundary conditions.

We should make sure that we only build these structures for the boundary faces that will at some point receive a condition, unless you think that is not possible. Maybe we should distinguish between "no condition now" and "no condition ever" in the type enum?

I will review the opm-simulators PR now with a view to this.

**Comment by totto82** [comment]
Thanks for the review. I added a if(nonTrivialBoundaryConditions). That should take care of the performance overhead if you dont have boundary conditions.

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

**Review by atgeirr — APPROVED** [review]

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 update_data please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 opm-tests=1089 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 update_data please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4912 opm-tests=1089 please

**Pull request merged**